### PR TITLE
Pre-commit hook to check mrns and csvs

### LINKED ISSUE #1270: pre-commit hooks check committed code for PHI
## What and why
It can be difficult to ensure you do not commit PHI, and we should automate everything!

## Solution(s)
`pre-commit` hook?

### PR DESCRIPTION
**Pull request by epalto**
## Issues closed
Closes #1270 

## Changelog
- [x] Added a git commit hook that checks PHI

## Checklist
- [x] The following are complete:
    - Added or revised tests if appropriate.
    - If new packages are installed, `requirements.txt` is updated.
    - If documentation should be updated, a new issue was made in `ml4c3-wiki`.
    - The PR name describes its content and does not contain "Ref #".
    - The PR does not add PHI.

### CODE REVIEW

**Comment by epalto** [comment]
The current behaviour of the hook is the following:
* It will check for numbers made of 5 to 8 digits
* It will discard any number that starts with at least 4 repetitions of the same digit.

Here is an example of how the regex works: https://regex101.com/r/4YY2YA/1

**Comment by StevenSong** [comment]
why max out at 8 digits?

also is it possible to additionally check for a comment that indicates its not an mrn? akin to `#isort: skip`?

**Comment by erikr** [comment]
> why max out at 8 digits?

Per https://github.com/aguirre-lab/ml4c3/wiki/Patient-identifiers BWH MRNs are 8 digits.

But that does not necessarily mean 8 should the max here.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ exclude: '^(.mypy_cache|.github)'
 
 repos:
 
+
```
Comment (erikr) [inline .pre-commit-config.yaml:8]:
```suggestion
```

**Comment by epalto** [comment]
Update:
* Now it looks for numbers of more than 4 digits (no upper limit)
* Two options on comments:
   - `sdc: nsd`: it ignores the line
   - `sdc: nsd-stop`:  it stops looking for sensitive data on the rest of the file

**Review by erikr — APPROVED** [review]

**Pull request merged**